celery/security/__init__.py

```diff
@@ -1,11 +1,25 @@
 """Message Signing Serializer."""
+from typing import TYPE_CHECKING, Iterable, Optional
+
 from kombu.serialization import disable_insecure_serializers as _disable_insecure_serializers
 from kombu.serialization import registry
 
 from celery.exceptions import ImproperlyConfigured
 
 from .serialization import register_auth  # : need cryptography first
 
+if TYPE_CHECKING:
+    import sys
+
+    from celery.app.base import Celery, Settings
+
+    if sys.version_info < (3, 8):
+        from typing_extensions import Literal
+    else:
+        from typing import Literal
+
+    _Serializer = Literal["json", "msgpack", "yaml", "pickle"]
```

Comment:
Here is kind of tricky, because the user can also add custom serializers. Meaning that the annotation should be `str` instead. I'm open to change it.

Also, not related, but I have two `_Serializer` in this PR, on different files. I accept suggestion on where to put them, if duplicated is not wanted.